File: playwright_script.js
const { chromium } = require('playwright');

(async () => {
  const browser = await chromium.launch();
  const page = await browser.newPage();

  try {
    console.log('Navigating to Google...');
    await page.goto('https://www.google.com');
    console.log('Page loaded. Taking screenshot...');
    await page.screenshot({ path: 'google_screenshot.png' });
    console.log('Screenshot saved as google_screenshot.png');
  } catch (error) {
    console.error('An error occurred:', error);
  } finally {
    await browser.close();
    console.log('Browser closed.');
  }
})();

File: playwright_script.py
import asyncio
from playwright.async_api import sync_playwright

async def run():
    async with sync_playwright() as p:
        browser = await p.chromium.launch()
        page = await browser.new_page()

        try:
            print('Navigating to Google...')
            await page.goto('https://www.google.com')
            print('Page loaded. Taking screenshot...')
            await page.screenshot(path='google_screenshot.png')
            print('Screenshot saved as google_screenshot.png')
        except Exception as e:
            print(f'An error occurred: {e}')
        finally:
            await browser.close()
            print('Browser closed.')

if __name__ == '__main__':
    asyncio.run(run())
